editor/src/core/shared/element-path-tree.ts

```diff
@@ -159,67 +167,93 @@ function getReorderedPaths(
   metadata: ElementInstanceMetadataMap,
   missingParents: ElementPath[],
 ): ElementPath[] {
-  const elementsToReorder = original.filter((path) => {
-    const element = MetadataUtils.findElementByElementPath(metadata, path)
-    if (element == null) {
-      return false
-    }
-    return (
-      MetadataUtils.isConditionalFromMetadata(element) ||
-      MetadataUtils.isExpressionOtherJavascriptFromMetadata(element) ||
-      MetadataUtils.isJSXMapExpressionFromMetadata(element) ||
-      MetadataUtils.isIdentifierFromMetadata(element) ||
-      MetadataUtils.isPropertyAccessFromMetadata(element) ||
-      MetadataUtils.isElementAccessFromMetadata(element)
-    )
+  const pathsToBeReordered = original.filter((path) => {
+    // Omit elements that have a missing parent.
+    return !missingParents.some((parentPath) => EP.isDescendantOf(path, parentPath))
   })
-  const pathsToBeReordered = original.filter(
-    (path) =>
-      !elementsToReorder.some((other) => EP.pathsEqual(other, path)) && // omit conditionals, that will be reordered
-      !missingParents.some((parentPath) => EP.isDescendantOf(path, parentPath)), // omit elements that have a missing parent
-  )
-  return elementsToReorder.reduce((paths, path) => {
+
+  return original.reduce((paths, path) => {
     const index = getReorderedIndexInPaths(paths, metadata, path)
     if (index === 'do-not-reorder') {
       return paths
     }
-    const before = paths.slice(0, index)
-    const after = paths.slice(index)
+    const pathsWithout = paths.filter((pathEntry) => !EP.pathsEqual(pathEntry, path))
```

Comment:
Minor, but if getReorderedIndexInPaths would return the original index of path too, we would not have to search for it again with this filter

Reply:
Hmmm, it'll make this logic a bit funkier, I'll leave it for the moment.